packages/wallet/ui/src/components/Proposal.jsx

```diff
@@ -56,12 +96,30 @@ const cmp = (a, b) => {
 const sortedEntries = entries =>
   Object.entries(entries).sort(([kwa], [kwb]) => cmp(kwa, kwb));
 
-const Proposal = ({ offer }) => {
+const Proposal = ({ offer, purses }) => {
   const {
-    proposalForDisplay: { give = {}, want = {}, arguments: args } = {},
+    proposalForDisplay,
+    proposalTemplate,
     invitationDetails: { fee, feePursePetname, expiry } = {},
   } = offer;
 
+  let give = {};
+  let want = {};
+  let args;
+  let hasDisplayInfo = false;
+  if (proposalForDisplay) {
+    give = proposalForDisplay.give ?? {};
+    want = proposalForDisplay.want ?? {};
+    args = proposalForDisplay.arguments;
+    hasDisplayInfo = true;
+  } else if (proposalTemplate) {
+    give = proposalTemplate.give ?? {};
+    want = proposalTemplate.want ?? {};
+    args = proposalTemplate.arguments;
+  }
```

Comment:
is it intended that `! proposalForDisplay && ! proposalTemplate` result in `give = {}` etc? Consider including an explanation.
```suggestion
  } else {
    // leave default values
  }
```

Reply:
Yes, added an explanation